### PR TITLE
Add Dogma code style linter

### PR DESCRIPTION
**Pull request by lpil**

### CODE REVIEW

File: lib/slime/tree.ex [hunk 1]
```diff
@@ -1,39 +1,73 @@
 defmodule Slime.Tree do
+  @moduledoc """
+  Build a tree from a series of Slim lines.
```
Comment (doomspork) [inline lib/slime/tree.ex]: Should this be "Slime" instead of "Slim"?

Comment (lpil) [inline lib/slime/tree.ex]: I guess so!

Comment (doomspork) [inline lib/slime/tree.ex]: There's a few other spots in the comments where Slim is used as well

Comment (lpil) [inline lib/slime/tree.ex]: I'll go through them.

**Comment by doomspork** [comment]
I'm excited for Dogma!  I'd like to be better about incorporating it in some of my other projects.  Looking forward to seeing how you set it up.

**Comment by lpil** [comment]
Oops, I broke some things with my regex. Will fix tomorrow.

**Comment by doomspork** [comment]
No worries.  Thought I'd let you know I picked up [slime-lang.com](http://slime-lang.com).  The DNS has been updated to point to the GitHub pages.

**Comment by lpil** [comment]
Excellent, thanks :)

File: lib/slime/compiler.ex [hunk 2]
```diff
@@ -33,42 +41,58 @@ defmodule Slime.Compiler do
...
 
-  defp render_branch(%{type: :doctype, content: text}), do: text
-  defp render_branch(%{type: :text, content: text}), do: text
-  defp render_branch(%{} = branch) do
-    opening = branch.attributes
-              |> Enum.map(fn {k, v} -> render_attribute(k, v) end)
-              |> Enum.join
-              |> open(branch)
 
+  defp render_branch(%{type: :doctype, content: text}) do
```
Comment (Rakoth) [inline lib/slime/compiler.ex]: Is it dogma forcing `do-end` style over `do:` style? I don't see why we should not use `do:` style.

File: lib/slime/parser.ex [hunk 3]
```diff
@@ -10,16 +14,40 @@ defmodule Slime.Parser do
...
-    "frameset":       "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Frameset//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-frameset.dtd\">",
-    "html":           "<!DOCTYPE html>",
-    "mobile":         "<!DOCTYPE html PUBLIC \"-//WAPFORUM//DTD XHTML Mobile 1.2//EN\" \"http://www.openmobilealliance.org/tech/DTD/xhtml-mobile12.dtd\">",
-    "strict":         "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">",
-    "transitional":   "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">",
-    "xml ISO-8859-1": "<?xml version=\"1.0\" encoding=\"iso-8859-1\" ?>",
-    "xml":            "<?xml version=\"1.0\" encoding=\"utf-8\" ?>"]
+    "1.1":
+      ~S[<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" ]
+      <> ~S["http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">],
```
Comment (Rakoth) [inline lib/slime/parser.ex:19]: Is there a way to sometimes allow long lines such as here?

File: test/renderer_test.exs [hunk 4]
```diff
@@ -45,48 +69,51 @@ defmodule RendererTest do
...
+  </body>
+  </html>
+  """ |> String.replace("\n", "")
 
   test "render inline html" do
     articles = [%{name: "Art 1", desc: "Desc 1"}, %{name: "Art 2", desc: "Desc 2"}]
     assert render(@inline_slim, articles: articles) == @inline_html
   end
 
-  test "render lines with 'do'" do
```
Comment (Rakoth) [inline test/renderer_test.exs:55]: Where this test goes?

File: lib/slime/compiler.ex (hunk 2 above)
Comment (lpil) [inline lib/slime/compiler.ex]: It's not, but I find do: hard to read for anything but the shortest of bodies.

File: lib/slime/parser.ex (hunk 3 above)
Comment (lpil) [inline lib/slime/parser.ex:19]: It's very close in Dogma master, but I've not finished it yet.

File: test/renderer_test.exs (hunk 4 above)
Comment (lpil) [inline test/renderer_test.exs:55]: Think it might have got lost in the last rebase on master. I'll put it back.

File: lib/slime/compiler.ex (hunk 2 above)
Comment (lpil) [inline lib/slime/compiler.ex]: Oh, this is a short body. I guess I just decided I wanted to change it. I can put it back if you like.

File: lib/slime/parser/attributes_keyword.ex [hunk 5]
```diff
@@ -20,7 +25,9 @@ defmodule Slime.Parser.AttributesKeyword do
     Enum.reduce(merge_rules, keyword_list, fn ({attr, join}, result) ->
       case Keyword.get_values(result, attr) do
         [] -> result
-        values -> Keyword.put(result, attr, merge_attribute_values(values, join))
+        values ->
+        values = merge_attribute_values(values, join)
+        Keyword.put(result, attr, values)
```
Comment (lpil) [inline lib/slime/parser/attributes_keyword.ex]: Need to fix this intendation

File: lib/slime/renderer.ex [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
+  @moduledoc """
+  Transform Slime templates into HTML.
+  """
   import Slime.Parser
   import Slime.Compiler
   import Slime.Tree
 
   @doc """
-  Compile slim template to valid EEx HTML.
+  Compile slime template to valid EEx HTML.
```
Comment (lpil) [inline lib/slime/renderer.ex]: Should probably use a capital S.

File: lib/slime/compiler.ex (hunk 2 above)
Comment (Rakoth) [inline lib/slime/compiler.ex]: It is up to you then.

Comment (doomspork) [inline lib/slime/compiler.ex]: I prefer the one-liners (`do:`) for short bodies.

**Comment by doomspork** [comment]
Thanks for your help @Rakoth :grinning:

**Comment by lpil** [comment]
Good to go?

File: lib/slime/compiler.ex [hunk 7]
```diff
@@ -33,42 +41,54 @@ defmodule Slime.Compiler do
...
+
   defp open(_, %{type: :eex, content: code, attributes: attrs}) do
     inline = if attrs[:inline], do: "=", else: ""
     "<%#{inline} #{code} %>\r"
   end
-
-  defp open(_, %{type: :html_comment}), do: "<!--"
-  defp open(_, %{type: :ie_comment, content: conditions}), do: "<!--[#{conditions}]>"
+  defp open(_, %{type: :html_comment}) do
+    "<!--"
```
Comment (doomspork) [inline lib/slime/compiler.ex:63]: @lpil I noticed you changed some of the `do ... end` back to `do:`, any reason not to change this one and the one below?  Same question for `close/1` on Line 79.

**Comment by doomspork** [comment]
Only one last last/feedback about one-liners.  Want to make sure we're consistent :grinning:

File: lib/slime/compiler.ex (hunk 7 above)
Comment (lpil) [inline lib/slime/compiler.ex:63]: Nope, I just missed them, this PR is a bit too big to keep in your head at once. I'll change the others too.

Comment (doomspork) [inline lib/slime/compiler.ex:63]: Hey Lousi,  I am traveling all day so I won't have a chance to look at this for a while.  Once you've incorporated those changes I think this is good to merge 👍

**Pull request merged**